lib/putasset.js

```diff
@@ -30,8 +38,66 @@ module.exports = async (token, {owner, repo, tag, filename}) => {
         size,
         url,
     });
+
+    return [
+        "https://github.com",
+        owner,
+        repo,
+        "releases",
+        "download",
+        tag, 
+        path.basename(filename)
+    ].join("/");
+
 };
 
+async function deleteReleaseAssetIfExist({ owner, repo, tag, filename, force }) {
+    const file_basename = path.basename(filename);
+
+    const asset_id = await (async () => {
+
```

Comment:
What purpose of this self invoking function?

Reply:
It's an async closure. it's purpose is to return the ``asset_id: number | undefined`` of the asset that would be overwritten.
It return ``undefined`` if there is no such asset.
I am not keen to refactor this bit.